src/score_match.cpp

```diff
@@ -123,8 +125,12 @@ float recursive_match(const MatchInfo &m,
       }
 
       // Apply a severe penalty if the case doesn't match.
-      // This should always hoist the exact case matches above all others.
-      if (m.smart_case && m.needle[needle_idx] != m.haystack[j]) {
+      // This should always hoist the exact matches above case insensitive or
```

Comment:
Nice, this is what I was worried about with this approach, but as long as exact matches win I think it makes sense.